examples/solitaire/src/SolitairePuzzle.tsx

```diff
@@ -106,29 +109,41 @@ export const Puzzle = ({
         cards.push(column);
         currentIndex += i;
       }
+
+      const newState: HistoryState = {
+        flippedCards: [],
+        stockCards: shuffledCards.slice(28), 
+        suitCards: state.suitCards,
+        tableauCards: cards, 
+        wasteCards: state.wasteCards,
+      };
+      setGameHistory((prev) => {
+        const state = [...(prev ?? []), newState];
+        onProgress?.(JSON.stringify(state));
+        return state;
+      });
+
       return {
         ...prev,
         stockCards: shuffledCards.slice(28),
         tableauCards: cards,
       };
-    });
+    }); 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
```

Comment:
`state` needs to be added as a dependency here

```suggestion
  }, [state]);
```